### PR TITLE
bpo-40286: Use random.randbytes() in tests

### PR DESCRIPTION
**Pull request by vstinner**
https://bugs.python.org/issue40286

### CODE REVIEW

**Comment by vstinner** [comment]
cc @serhiy-storchaka

File: Lib/test/test_tarfile.py [hunk 1]
```diff
@@ -386,7 +386,7 @@ def test_null_tarfile(self):
     def test_ignore_zeros(self):
         # Test TarFile's ignore_zeros option.
         # generate 512 pseudorandom bytes
-        data = Random(0).getrandbits(512*8).to_bytes(512, 'big')
+        data = Random(0).randbytes(512)
```
Comment (serhiy-storchaka) [inline Lib/test/test_tarfile.py:389]: I think it is better to use the global `randbytes()`.

**Review by serhiy-storchaka — COMMENTED** [review]
Huh! I just have wrote almost identical patch!

There is yet one place in test_zlib:
```diff
@@ -829,13 +828,9 @@ def genblock(seed, length, step=1024, generator=random):
     """length-byte stream of random data from a seed (in step-byte blocks)."""
     if seed is not None:
         generator.seed(seed)
-    randint = generator.randint
-    if length < step or step < 2:
-        step = length
-    blocks = bytes()
-    for i in range(0, length, step):
-        blocks += bytes(randint(0, 255) for x in range(step))
-    return blocks
+    if length > step > 1:
+        length += -length % step
+    return generator.randbytes(length)
```

File: Lib/test/test_tarfile.py (hunk 1 above)
Comment (vstinner) [reply Lib/test/test_tarfile.py:389]: It uses a seed of 0. I would prefer to use a temporary instance, rather than modifying random._inst.

**Comment by vstinner** [comment]
> Huh! I just have wrote almost identical patch!

He he, great minds think alike ;-)

**Comment by vstinner** [comment]
> There is yet one place in test_zlib: (...)

Oh right, I also updated this code.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**